pkg/proc/disasm_amd64_arch.go

```diff
@@ -1,3 +1,5 @@
+// +build amd64
```

Comment:
I think this file should be left with its name `disasm_amd64.go`. Instead add this note:

```
// TODO: disassembler support should be compiled in unconditionally, instead of being decided by the build-target architecture, and be part of the Arch object instead.
```